src/objects/box.ts

```diff
@@ -2,22 +2,31 @@ import { endOfHour, startOfHour } from 'date-fns'
 import { Viewport } from 'pixi-viewport'
 import { Sprite, Texture } from 'pixi.js'
 import { emitter } from '@/objects/events'
-import { scaleX, scaleY } from '@/objects/scales'
+import { Scales, waitForScales } from '@/objects/scales'
+import { waitForViewport } from '@/objects/viewport'
 
 let sprite: Sprite | null = null
 
-export function startBox(): void {
-  emitter.on('viewportCreated', createBox)
-  emitter.on('scaleXUpdated', renderBox)
+export async function startBox(): Promise<void> {
+  const viewport = await waitForViewport()
+  createBox(viewport)
+
+  const scales = await waitForScales()
+  renderBox(scales)
+
```

Comment:
Yeah I think this pattern is turning out nicely

Reply:
Yeah I'm loving this so far